python/ucxx/ucxx/_lib_async/continuous_ucx_progress.py

```diff
@@ -24,12 +29,19 @@ def __init__(self, worker, event_loop):
         self.event_loop = event_loop
         self.asyncio_task = None
 
-    def __del__(self):
-        if self.asyncio_task is not None:
-            # FIXME: This does not work, the cancellation must be awaited.
-            # Running with polling mode will always cause
-            # `Task was destroyed but it is pending!` errors at ucxx.reset().
-            self.asyncio_task.cancel()
+        event_loop_close = self.event_loop.close
+
+        def _event_loop_close(*args, **kwargs):
+            if not self.event_loop.is_closed() and self.asyncio_task is not None:
+                try:
+                    self.asyncio_task.cancel()
+                    self.event_loop.run_until_complete(self.asyncio_task)
```

Comment:
*suggestion*: should we set `self.asyncio_task = None` after running until complete?

Reply:
OK, so the idea is that you don't have control over who is closing the event loop, so you instead hook into `close` and this task cancels itself during event loop closing/teardown?

Reply:
> _suggestion_: should we set `self.asyncio_task = None` after running until complete?

That's a good idea, done in https://github.com/rapidsai/ucxx/pull/116/commits/279cb4c0b4d11fa9df6df29aaf4083e27e30018e .

> OK, so the idea is that you don't have control over who is closing the event loop, so you instead hook into `close` and this task cancels itself during event loop closing/teardown?

Exactly, there doesn't seem to be another way, since we have no control of whether the user will close the loop before resetting UCXX.